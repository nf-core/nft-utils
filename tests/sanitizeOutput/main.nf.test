nextflow_process {

    name "Test sanitizeOutput"
    script "./modules/local/test/main.nf"
    process "TEST"
    tag "sanitizeOutput"

    test("Default sanitize output") {
        when {
          process {
            """
            input[0] = [id:"test"]
            """
          }
        }


        then {
            assert process.success
            assert snapshot(sanitizeOutput(process.out, unstableKeys:["zip"])).match()
        }
    }
}
