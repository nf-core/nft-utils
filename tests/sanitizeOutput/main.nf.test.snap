{
    "Default sanitize output": {
        "content": [
            {
                "html": [
                    [
                        {
                            "id": "test"
                        },
                        "test_fastqc.html:md5,24854f9d4c2bc9e908174cabefb1cf07"
                    ]
                ],
                "id": [
                    "test"
                ],
                "zip": [
                    [
                        {
                            "id": "test"
                        },
                        "test_fastqc.zip"
                    ]
                ],
                "zip_only": [
                    "test_fastqc.zip:md5,b14838a8cfd436e61fbf159d7255cc59"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-11T14:51:48.072281016"
    }
}